packages/flutter/lib/src/services/restoration.dart

```diff
@@ -294,23 +294,53 @@ class RestorationManager extends ChangeNotifier {
     _bucketsNeedingSerialization.remove(bucket);
   }
 
-  void _doSerialization() {
+  Future<void> _doSerialization() {
+    if (!_serializationScheduled) {
+      return Future<void>.value(null);
```

Comment:
just make the function async and `return;`.

Reply:
though... why do you not return the same future as the other call to doSerialization?
Or throw? Seems weird to do nothing when called redundantly, and return a future that completes before the earlier "real" call...